backend/src/api/endpoints/search.py

```diff
@@ -77,7 +81,11 @@ async def search_code_stats(
     user_ids: List[int],
     sdoc_ids: List[int],
     sort_by_global: bool = False,
+    authz_user: AuthzUser = Depends(),
 ) -> List[SpanEntityStat]:
+    authz_user.assert_in_same_project_as(Crud.CODE, code_id)
+    authz_user.assert_in_same_project_as_many(Crud.SOURCE_DOCUMENT, sdoc_ids)
```

Comment:
wie performant ist das? sdoc_ids sind teilweise 10k+ ids :/

Reply:
Mit der neuen `assert_in_same_project_as_many` methode sollte es in Ordnung sein, es werden zwar alle sdocs geladen, aber nur über eine einzelne query. Wenn du möchtest, kann ich dir noch genaue Messungen machen :)